sdks/python/apache_beam/transforms/core.py

```diff
@@ -2587,6 +2587,18 @@ def add_input_types(transform):
             "or CombineGlobally().as_singleton_view() to get the default "
             "output of the CombineFn if the input PCollection is empty.")
 
+      # return the error for this ill-defined streaming case
+      if not pcoll.is_bounded and not pcoll.windowing.is_default():
+        raise ValueError(
```

Comment:
1. Is the this behavior applicable only for CombineGlobally or also for CombinePerKey / CombineValues ? Perhaps we should generalize the wording here if other tests also fail in this scenario.

2. For my understanding, how is the default value defined if we do use the default trigger?

3. Should this wording be more appropriate: 

```
"When combining elements in unbounded collections with non-default windowing strategy, you must explicitly specify how to define the combined result of an empty window. "

"Please use CombineGlobally().without_defaults() to output "
            "an empty PCollection if the input PCollection is empty, "
            "or CombineGlobally().as_singleton_view() to get the default "
            "output of the CombineFn if the input PCollection is empty.")
```            
4. If users specify `as_singleton_view`, do we need to check that  underlying CombineFn `.has_defaults()`? Is a meaningful error thrown  if it doesn't?